enhancements/installer/agent-based-installation-in-hive.md

```diff
@@ -0,0 +1,647 @@
+which rules out the use of an "alpha" API version due to its insufficient level
+of commitment.
+
+New CRDs will have version "v1beta1" during the OpenShift 4.8 timeline (though
+they will be distributed as part of RHACM). This will enable them to be
+supported in production while allowing greater flexibility for changes than if
+they were released as "v1".
+
+The agent-specific portions of the hive ClusterDeployment will inherit the
+ClusterDeployment's current version of "v1".
+
+#### Agent REST API Auth
+
+Agents will communicate with the assisted service using the non-kubernetes REST
+API that is currently in use by the SaaS. The backend assisted service will be
+listening on a socket that is routable from anywhere Agents are expected to
+run, so it needs to be secured from other potential actors on the network.
+
+When an Agent connects to report its existence, send logs, send inspection
+data, etc. it will need to authenticate.
+
+An authentication artifact will be embedded into the discovery ISO for use by
+the agent. As long as the user protects the contents of the discovery ISO
+itself, they can be confident that any client interacting with the backend
+service's REST API is the expected Agent running as part of the live discovery
+ISO. Details of the artifact are still being decided.
+
+#### Other REST API Access
+
+Where the assisted service has capabilities which are not (yet) expressed
+through kubernetes APIs, users may be tempted to access those capabilities
+directly via the assisted servuce REST API, thereby taking a dependency on the
+capability and its REST API. This would limit our ability to evolve those APIs
+in future.
+
+To mitigate this, the assisted service's REST API, except for the Agent API as
+described above, will limited to internal use only. It may be limited through
+authorization mechanisms or by blocking entire URL path space from external
+access.
+
+## Design Details
+
+### Open Questions
+
+#### Can we use sqlite?
+
+It would be advantageous to use sqlite for the database especially on
+stand-alone clusters, so that it is not necessary to deploy and manage
+an entire RDBMS, nor ship and support its container image. The [gorm library
+supports sqlite](https://gorm.io/docs/connecting_to_the_database.html#SQLite),
+but it is not clear if assisted service is compatible. In particular, the [use
+of FOR
+UPDATE](https://github.com/openshift/assisted-service/blob/e70af7dcf59763ee6c697fb409887f00ab5540f5/pkg/transaction/transaction.go#L8)
+might be problematic.
+
+For Hub clusters doing multi-cluster creation and management, there is an
+expectation that persistent storage availability and scale concerns will be a
+better fit for running a full RDBMS.
+
+A suggestion has been made that in situations where there is a single process
+running the assisted service, locking can happen in-memory instead of in the
+database. Further analysis is required.
```

Comment:
Would a single instance of the service be sufficient to meet the scaling requirements?

Reply:
It should.  Concurrent ISO downloads might be a bottleneck at some point and we already have plans to scale that separately.